src/Our.Umbraco.InnerContent/Web/UI/App_Plugins/InnerContent/js/innercontent.controllers.js

```diff
@@ -350,8 +407,8 @@ angular.module("umbraco").factory('innerContentService', [
 
         var self = {};
 
-        var getScaffold = function (contentType) {
-            return icResources.getContentTypeScaffoldByGuid(contentType.icContentTypeGuid).then(function (scaffold) {
+        var getScaffold = function (contentType, blueprintId) {
+            return icResources.getContentTypeScaffoldByGuid(contentType.icContentTypeGuid, blueprintId).then(function (scaffold) {
```

Comment:
@mattbrailsford If we're going to have 2 controller methods (one for Guid, the other for BlueprintId), then how should I handle this here? Given the return method is from `getContentTypeScaffoldByGuid`'s promise?

Would it be to separate out the contents of the `then(function)`?

Reply:
yea, put the function in a local variable beforehand, then use it in either `then` call depending on which endpoing you need to call